osf/utils/fields.py

```diff
@@ -12,7 +12,7 @@ class LowercaseCharField(models.CharField):
     def get_prep_value(self, value):
         value = super(models.CharField, self).get_prep_value(value)
         if value is not None:
-            value = value.lower()
+            value = str(value).lower()
```

Comment:
Yeah, super calls [`to_python`](https://github.com/django/django/blob/1.9/django/db/models/fields/__init__.py#L1105), which checks to see if it's a string and then makes it a string if it isn't ([smart_text](https://github.com/django/django/blob/1.9/django/utils/encoding.py#L32) which calls [force_text](https://github.com/django/django/blob/1.9/django/utils/encoding.py#L58)). This is django's default behavior. If you want to prevent integers being cast you'd have to write your own validator and put it on the field. I'm not 100% sure but I'm fairly confident that if you did a SQL update with an integer into a varchar PostgreSQL would coerce it for you.